src/framework/audio/common/workmode.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore Limited and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "workmode.h"
+
+#include "global/settings.h"
+
+#include "muse_framework_config.h"
+
+using namespace muse;
+using namespace muse::audio;
+
+workmode::Mode workmode::m_mode = workmode::Mode::Undefined;
+workmode::Mode workmode::m_desiredMode = workmode::Mode::Undefined;
+
+static const Settings::Key WORKMODE_KEY("audio", "dev/workmode");
+
+void workmode::load()
+{
+    int val = settings()->value(WORKMODE_KEY).toInt();
+    if (val > 0 && val <= 3) {
+        m_mode = static_cast<Mode>(val);
+    } else {
+        m_mode = defaultMode();
+    }
+
+    m_desiredMode = m_mode;
+}
+
+workmode::Mode workmode::defaultMode()
+{
+    return static_cast<Mode>(MUSE_MODULE_AUDIO_WORKMODE);
```

Comment:
Now that it's done at runtime instead of compile time, there isn't really a reason anymore to get this from CMake, so we could make this cpp method the source of truth